hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/AsyncClusteringService.java

```diff
@@ -47,7 +47,7 @@ public abstract class AsyncClusteringService extends HoodieAsyncTableService {
   private static final Logger LOG = LoggerFactory.getLogger(AsyncClusteringService.class);
   private final int maxConcurrentClustering;
   protected transient HoodieEngineContext context;
-  private transient BaseClusterer clusteringClient;
+  private final transient BaseClusterer clusteringClient;
```

Comment:
I think the transient variables cannot be final if this is for serialization and deserialization in Spark.

Reply:
See https://stackoverflow.com/questions/2968876/final-transient-fields-and-serialization

Reply:
hmmm. how is the CI passing.. let me dig more.